internal/go_repository.bzl

```diff
@@ -88,7 +88,7 @@ def _go_repository_impl(ctx):
     generate = ctx.attr.build_file_generation == "on"
     if ctx.attr.build_file_generation == "auto":
         generate = True
-        for name in ["BUILD", "BUILD.bazel", ctx.attr.build_file_name]:
+        for name in depset(["BUILD", "BUILD.bazel"] + ctx.attr.build_file_name.split(",")).to_list():
```

Comment:
Depsets are suprisingly complicated, and deduplication isn't really important here. Maybe just:

```
["BUILD", "BUILD.bazel"] + ctx.attr.build_file_name.split(",")
```